### PR TITLE
Fix JNLP for more cases

### PR DESCRIPTION
**Pull request by tobias-**
* user is now set
* script is fixed so that root can be used

[Review on Reviewable](https://reviewable.io/reviews/jenkinsci/docker-plugin/293)

### CODE REVIEW

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerJNLPLauncher.java [hunk 1]
```diff
@@ -161,7 +161,10 @@ public void launch(SlaveComputer computer, TaskListener listener) throws IOExcep
 
     @Override
     public ComputerLauncher getPreparedLauncher(String cloudId, DockerTemplate template, InspectContainerResponse containerInspectResponse) {
-        return new DockerComputerJNLPLauncher(getJnlpLauncher());
+        DockerComputerJNLPLauncher dockerComputerJNLPLauncher = new DockerComputerJNLPLauncher(getJnlpLauncher());
+        DockerComputerJNLPLauncher launcher = (DockerComputerJNLPLauncher) template.getLauncher();
+        dockerComputerJNLPLauncher.setUser(launcher.getUser());
+        return dockerComputerJNLPLauncher;
```
Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerJNLPLauncher.java]: It should be implemented in other way, using clone interfaces like SSHLauncher that need to be fixed also. It was a temp hack.

File: docker-plugin/src/main/resources/com/nirima/jenkins/plugins/docker/launcher/DockerComputerJNLPLauncher/init.sh [hunk 2]
```diff
@@ -38,9 +38,9 @@ if [ ! -z "$COMPUTER_SECRET" ]; then
...
 
-RUN_CMD=""
+RUN_CMD="java -jar ${JENKINS_HOME}/slave.jar $RUN_OPTS"
 if [ ! -z "$JENKINS_USER" ] && [ x"$JENKINS_USER" != "xroot" ]; then
- RUN_CMD="su - $JENKINS_USER -c"
+ RUN_CMD="su - $JENKINS_USER -c '$RUN_CMD'"
 fi
 
-$RUN_CMD "java -jar slave.jar $RUN_OPTS"
+$RUN_CMD
```
Comment (KostyaSha) [inline docker-plugin/src/main/resources/com/nirima/jenkins/plugins/docker/launcher/DockerComputerJNLPLauncher/init.sh]: It was temp hack.

**Comment by KostyaSha** [comment]
Thanks for PR, but i don't plan ship JNLP launcher :atm: until i implement integration tests. (It was to hard to add stability with SSH launching).

**Comment by hoeghh** [comment]
I think its a shame. The PR worked, and didnt need a SSH connection (and therefore no credentials either).

**Pull request merged**